components/ui/atoms/list-card-item/CardFavoriteButton.tsx

```diff
@@ -0,0 +1,31 @@
+import styles from "./CardItem.module.css";
+import beforeClickImage from "@/public/assets/images/icons/starIcon.svg";
+import afterClickImage from "@/public/assets/images/icons/starActiveIcon.svg";
+import Image from "next/image";
+import { useState } from "react";
+
+const CardFavoriteButton = () => {
+  const [isFavorite, setIsFavorite] = useState(false);
+
+  const handleActiveFavoriteClick = (e: any) => {
+    // 버블링으로 인해 favorite 버튼을 누르면, Link 태그가 클릭되어져 버립니다.
+    // preventDefault로 해결하였는데, 다른 더 좋은 방법 있을까요?
+    e.preventDefault();
+    setIsFavorite(!isFavorite);
```

Comment:
```suggestion
    setIsFavorite((currentValue)=>!curruentValue);
```
이렇게 작성해주시는게 조금 더 안전한 방법이에요~ 원본코드에서는 코드의 시점에 따라서 내가 원하지 않는 값을 참조할수도 있는데 제가 작성한 코드에서는 Clousre의 특성으로 원하는 값으로 토글하는 함수를 작성할 수 있어요!

Reply:
af1f96ac09cb122d5357c28a8fa2e765e9f6ae94 호곡 useState를 저렇게 사용할 수 있는건 처음 알았어요! 관련해서 더 공부해 보아야겠습니다. 감사합니당